R/uvhydrograph-data.R

```diff
@@ -225,17 +225,56 @@ parseLabelSpacing <- function(data, info) {
   
   if (names(data) %in% c("series_corr", "series_corr_ref", "series_corr_up", "series_corr2")){
     limits <- info[[grep("lims_UV", names(info))]]
-    y_positions <- rep(limits$ylim[2], length(data[[1]]$time))
-    differences <- as.numeric(diff(data[[1]]$time))
-    if(length(differences) > 0) {
-      for (i in seq_len(length(differences))) {
-        if(abs(differences[i]) < 86400) {y_positions[i+1] <- y_positions[i]-(0.04*info$lims_UV$ylim[2])}
-        i <- i + 1
-      }
-      spacingInfo <- list(y=y_positions, label=seq(length(data[[1]]$time)))
-    } else {
-      spacingInfo <- list(y=y_positions, label=seq(length(data[[1]]$time)))
-    }
+    #Number of hours to offset labels by for display
+    hourOffset <- 4 * 60 * 60
+    #Width of one digit in hours
+    digitHours <- 4 * 60 * 60
+    #Total width of both bounding box left and right margins 
+    baseBoxSize <- 4 * 60 * 60
+    #Minimum space between the right side of a label box and the next correction line to not merge columns
+    minSpacerSize <- 2 * 60 * 60
+    #The percentage of the y-range to subtract each time we add a new label to a column
+    subtractor <- (limits$ylim[[2]] - limits$ylim[[1]]) * 0.065
+
+    #Save original order as label and re-order by time and then by label (descending)
+    corrs <- data[[1]] %>% select(time) %>% mutate(label = row_number()) %>% arrange(time, desc(label))
```

Comment:
I agree that this is pretty confusing and hard to follow as-is. I think pair programming sounds like a good idea. If I am able to make it into the office this afternoon I'll let you know. In the meantime I'll do some thinking about it and some tinkering with it and see if I can come up with some ideas to make it more recursive and less dplyr-dependent.